test/child_killing_test.rb

```diff
@@ -92,11 +98,26 @@ def assert_child_not_running(child_pid)
     old_job_per_fork = ENV['FORK_PER_JOB']
     begin
       ENV['FORK_PER_JOB'] = 'false'
-      worker_pid, child_pid, result = start_worker(0, true)
+      worker_pid, child_pid, result = start_worker_and_kill_term(0, true)
       assert_equal worker_pid, child_pid, "child_pid should equal worker_pid, since we are not forking"
       assert Resque.redis.lpop( 'sigterm-test:ensure_block_executed' ), 'post cleanup did not occur. SIGKILL sent too early?'
     ensure
       ENV['FORK_PER_JOB'] = old_job_per_fork
     end
   end
+
+  it 'sending signal to a forked child should not make a effect to its parent' do
+    worker_pid, child_pid = start_worker(0, false)
+
+    Process.kill('TERM', child_pid)
+    10.times do
+      assert_nil Process.waitpid(worker_pid, Process::WNOHANG) # should not die
+      sleep 0.1
+    end
+
+    Process.kill('TERM', worker_pid)
+    Process.waitpid(worker_pid)
+    result = Resque.redis.lpop('sigterm-test:result')
```

Comment:
it seems that LongRunningJob does `rpush`, this line is just for `lpop` it. Is it not needed?